### PR TITLE
Pass column names to `write_csv` instead of `table_metadata` pointer

### PR DESCRIPTION
**Pull request by vuule**
## Description
contributes to https://github.com/rapidsai/cudf/issues/6411
`write_csv` takes a pointer to `table_metadata` but only uses the column names.
This PR changes the API to directly take column names. This also aligns with `read_csv`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

File: cpp/include/cudf/io/csv.hpp [hunk 1]
```diff
@@ -1391,7 +1391,7 @@ class csv_writer_options {
    *
    * @return Optional associated metadata
    */
-  [[nodiscard]] table_metadata const* get_metadata() const { return _metadata; }
+  [[nodiscard]] std::vector<std::string> const& get_names() const { return _names; }
```
Comment (ttnghia) [inline cpp/include/cudf/io/csv.hpp:1394]: Need to update doxygen for this func.

File: cpp/include/cudf/io/csv.hpp [hunk 2]
```diff
@@ -1339,7 +1339,7 @@ class csv_writer_options {
   // string to use for values == 0 in INT8 types (default 'false')
   std::string _false_value = std::string{"false"};
   // Optional associated metadata
-  table_metadata const* _metadata = nullptr;
+  std::vector<std::string> _names;
```
Comment (ttnghia) [inline cpp/include/cudf/io/csv.hpp:1342]: Sorry I still missed doxygen for this variable too. Probably the last missed one :smile:

**Review by ttnghia — APPROVED** [review]

**Comment by vuule** [comment]
rerun tests

File: cpp/include/cudf/io/detail/csv.hpp [hunk 3]
```diff
@@ -46,14 +46,14 @@ table_with_metadata read_csv(std::unique_ptr<cudf::io::datasource>&& source,
...
- * @param metadata The metadata associated with the table
+ * @param column_names Column names for the output CSV
  * @param options Settings for controlling behavior
  * @param stream CUDA stream used for device memory operations and kernel launches.
  * @param mr Device memory resource to use for device memory allocation
  */
 void write_csv(data_sink* sink,
                table_view const& table,
-               const table_metadata* metadata,
+               std::vector<std::string> const& column_names,
```
Comment (PointKernel) [inline cpp/include/cudf/io/detail/csv.hpp:56]: Can this be replaced by span?

**Review by PointKernel — APPROVED** [review]
LGTM. Thanks!

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**